activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -591,31 +588,42 @@ def add_foreign_key(from_table, to_table, **options)
         def foreign_keys(table_name)
           scope = quoted_scope(table_name)
           fk_info = internal_exec_query(<<~SQL, "SCHEMA", allow_retry: true, materialize_transactions: false)
-            SELECT t2.oid::regclass::text AS to_table, a1.attname AS column, a2.attname AS primary_key, c.conname AS name, c.confupdtype AS on_update, c.confdeltype AS on_delete, c.convalidated AS valid, c.condeferrable AS deferrable, c.condeferred AS deferred, c.conkey, c.confkey, c.conrelid, c.confrelid
```

Comment:
This deletes `column` and `primary_key` columns, as they can be recovered from `conkey_names` and `confkey_names` for a single column key (the first element of each array)